packages/config/src/index.ts

```diff
@@ -95,10 +95,12 @@ class TruffleConfig {
     const current = this.normalize(this);
     const normalized = this.normalize(obj);
 
-    const newConfig = assignIn(
+    const newConfig = Object.assign(
       Object.create(TruffleConfig.prototype),
       current,
-      normalized
+      Object.getPrototypeOf(current),
+      normalized,
+      Object.getPrototypeOf(normalized)
```

Comment:
Order is meaningful here. If an object shadows its prototype what should the behavior be? Does the prototype have precedence over an object's own properties in this operation or vice versa?

```
function Foo (name) { this.name = name; }
Foo.prototype.times = "proto";


// prototype is the truthiest
const assignIn_1 = (o, ...sources) => {
  return Object.assign(o, ...sources.flatMap(s => [s, Object.getPrototypeOf(s)]))
}

// ownProperty is the truthiest
const assignIn_2 = (o, ...sources) => {
  return Object.assign(o, ...sources.flatMap(s => [Object.getPrototypeOf(s), s]))
}

console.log("object before proto");
const foo = new Foo('me');
console.log(assignIn_1({a: 'angst'}, foo)); // { a: 'angst', name: 'me', times: 'proto' }
foo.times = "from the shadows";
console.log(assignIn_1({a: 'angst'}, foo)); // { a: 'angst', name: 'me', times: 'proto' }


console.log("proto before object");
const boo = new Foo('me');
console.log(assignIn_2({a: 'angst'}, boo)); // { a: 'angst', times: 'proto', name: 'me' }
boo.times = "from the shadows";
console.log(assignIn_2({a: 'angst'}, boo)); // { a: 'angst', times: 'from the shadows', name: 'me' }
```